lib/core/src/chain_swap.rs

```diff
@@ -367,6 +367,28 @@ impl ChainSwapHandler {
             | ChainSwapStates::TransactionLockupFailed
             | ChainSwapStates::TransactionRefunded
             | ChainSwapStates::SwapExpired => {
+                // Zero-amount Receive Chain Swaps also get to TransactionLockupFailed when user locks up funds
+                let is_zero_amount = swap.payer_amount_sat == 0;
+                if matches!(swap_state, ChainSwapStates::TransactionLockupFailed) && is_zero_amount
+                {
+                    if let Err(e) = self
+                        .swapper
+                        .get_zero_amount_chain_swap_quote(&swap.id)
+                        .map(|quote| quote.to_sat())
+                        .and_then(|quote| {
+                            info!("Got quote of {quote} sat for swap {}", &swap.id);
+
+                            self.swapper
```

Comment:
Should we also update the swap payer_amount here?

Reply:
Makes sense, done: 35cb343b7b310e15f32bde5202f49c7fa4109a11